repair/row_level.cc

```diff
@@ -662,18 +662,30 @@ future<std::list<repair_row>> to_repair_rows_list(repair_rows_on_wire rows, sche
     co_return std::move(row_list);
 }
 
-void flush_rows(schema_ptr s, std::list<repair_row>& rows, lw_shared_ptr<repair_writer>& writer) {
+void flush_rows(schema_ptr s, std::list<repair_row>& rows, lw_shared_ptr<repair_writer>& writer, locator::effective_replication_map_ptr erm, bool small_table_optimization) {
     auto cmp = position_in_partition::tri_compare(*s);
     lw_shared_ptr<mutation_fragment> last_mf;
     lw_shared_ptr<const decorated_key_with_hash> last_dk;
+    bool enable_opt = erm && small_table_optimization;
+    auto* strat = enable_opt ? &erm->get_replication_strategy() : nullptr;
+    auto* tm = enable_opt ? &erm->get_token_metadata() : nullptr;
+    auto myip = enable_opt ? utils::fb_utilities::get_broadcast_address() : gms::inet_address();
     for (auto& r : rows) {
         thread::maybe_yield();
         if (!r.dirty_on_master()) {
             continue;
         }
+        const auto& dk = r.get_dk_with_hash()->dk;
+        if (enable_opt) {
+            // Check if the token is owned by the node
+            auto eps = strat->calculate_natural_endpoints(dk.token(), *tm).get0();
```

Comment:
nitpick: I wonder if we don't have a more efficient API for just checking if a token belongs to a node instead of constructing and looking up in a hash table. What does "nodetool cleanup" use on each token?
By the way, don't we see token in token order? In that case, we know that all tokens in a whole range belong to this node - it should be more efficient than checking each one individually as if it's a random-access lookup.

Reply:
Since the data can come from any token range from the whole token space, we could not calculate the token range to nodes mapping once and reuse it.

Reply:
Why is the `strat->calculate_natural_endpoints` method is called rather than `erm->get_natural_endpoints`?
The token metadata is retrieved from the erm so we can use the erm precalculated map (for vnodes) or get the map for tablets otherwise.
`strat->calculate_natural_endpoints` assumes vnodes and it doesn't work with tablets.

Cc @tgrabiec